sys/net/gnrc/network_layer/ipv6/nib/_nib-router.c

```diff
@@ -242,43 +320,14 @@ static gnrc_pktsnip_t *_build_ext_opts(gnrc_netif_t *netif,
  * this link. */
 static gnrc_pktsnip_t *_build_final_ext_opts(gnrc_netif_t *netif)
 {
-    gnrc_pktsnip_t *ext_opts = NULL;
-    _nib_offl_entry_t *entry = NULL;
-
     if (!IS_USED(MODULE_GNRC_IPV6_NIB_RIO) ||
         !IS_ACTIVE(CONFIG_GNRC_IPV6_NIB_ADD_RIO_IN_LAST_RA)) {
         return NULL;
     }
 
     DEBUG("nib: sending final RA on interface %u\n", netif->pid);
 
-    uint32_t now = evtimer_now_msec();
-    while ((entry = _nib_offl_iter(entry))) {
-
-        unsigned id = netif->pid;
-        if (_nib_onl_get_if(entry->next_hop) == id) {
-            continue;
-        }
-
-        if ((entry->mode & _PL) && (entry->flags & _PFX_ON_LINK)) {
-            DEBUG("nib: adding downstream subnet to RA\n");
-            uint32_t valid_ltime = (entry->valid_until == UINT32_MAX) ? UINT32_MAX :
-                                   ((entry->valid_until - now) / MS_PER_SEC);
-            gnrc_pktsnip_t *snip  = gnrc_ndp_opt_ri_build(&entry->pfx,
-                                                          entry->pfx_len,
-                                                          valid_ltime,
-                                                          NDP_OPT_RI_FLAGS_PRF_NONE,
-                                                          ext_opts);
-            if (snip != NULL) {
-                ext_opts = snip;
-            } else {
-                DEBUG_PUTS("nib: can't add RIO to RA - out of memory");
-                break;
-            }
-        }
-    }
-
-    return ext_opts;
+    return _add_rio(netif, NULL, false);
```

Comment:
> It's now also optional in normal RAs

Not optional here. But it also was not optional before. Did you forget to commit or are wee keeping it like that.
I mean since it is a separate RFC, optional would not harm I guess

Reply:
This is after a

```C
    if (!IS_USED(MODULE_GNRC_IPV6_NIB_RIO) ||
        !IS_ACTIVE(CONFIG_GNRC_IPV6_NIB_ADD_RIO_IN_LAST_RA)) {
        return NULL;
    }
```

Reply:
Ah ok I am blind ^^. Just a nitpick, but can we have the same pattern as in `_build_ext_opts`. Also the comment "sending final RA on interface" could be misinterpreted, because this function does not perform a send but just builds an option